test/benchmark_http_client_test.cc

```diff
@@ -276,8 +297,38 @@ TEST_F(BenchmarkClientHttpTest, BadContentLength) {
   };
 
   EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(0);
-  testBasicFunctionality(1, 1, 1);
+  testBasicFunctionality(1, 1, 1, request_generator);
   EXPECT_EQ(1, getCounter("http_2xx"));
 }
-
+TEST_F(BenchmarkClientHttpTest, MultipleRequestsDifferentPath) {
+  std::vector<HeaderMapPtr> requests_vector;
+  std::initializer_list<std::pair<std::string, std::string>> header1{{":scheme", "http"},
+                                                                     {":method", "GET"},
+                                                                     {":path", "/a"},
+                                                                     {":host", "localhost"},
+                                                                     {"Content-Length", "1313"}};
+  std::initializer_list<std::pair<std::string, std::string>> header2{{":scheme", "http"},
+                                                                     {":method", "GET"},
+                                                                     {":path", "/b"},
+                                                                     {":host", "localhost"},
+                                                                     {"Content-Length", "1313"}};
+  requests_vector.push_back(std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header1));
```

Comment:
returned_request_vector would be a better name for this to clarify what's that this is the out

Reply:
Done.